docs/modules/ROOT/pages/how-to-guides/proc_prefetching-dependencies-to-support-hermetic-build.adoc

```diff
@@ -0,0 +1,193 @@
+:_content-type: PROCEDURE
+
+[id="prefetching-dependencies-to-support-hermetic-build_{context}"]
+= Prefetching the package manager dependencies for the Hermetic build
+
+{ProductName} uses Cachi2 to prefetch the package manager dependencies for the <<supported-languages>>. Cachi2 is a wrapper around the package managers that makes it easy to prefetch and cache dependencies. It also helps generate a manifest of all dependencies included in your builds, ensuring transparency and maintainability.
+
+[#supported-languages]
+.Supported languages
+
+[cols="1,1"]
+|===
+|Language
+|Package Manager
+
+|Go
+|`gomod`
+
+|Python
+|`pip`
+|===
+
+== Enabling prefetch builds for `gomod`
+
+.Prerequisites
+* You have an upgraded build pipeline.
+* You have a `go.mod` file in your repository that lists all the the dependencies.
+
+.Procedures
+For each component that you want to build hermetically:
+
+. Navigate to your component's source code.
+. Open the *.tekton* directory and locate the *.yaml* files related to the *pull-request* and *push* processes.
+. *Set up dependency prefetching with Cachi2:* In both the `.yaml` files, add the following prefetch-input pipeline parameter.
+
++
+[source,yaml]
+----
+param:
+    -  name: prefetch-input
+       value: gomod <1>
+----
+<1> If the package manager lock file is located in the root directory of your project, enter `gomod`. Otherwise, if the package manager lock file is in a subdirectory, provide the path to the lock file in JSON format. For example,
+{"type": "{<package-manager>}", "path": "path-to-the-directory-which-contains-the-lockfile}"}
```

Comment:
You can resolve the `{<package-manager>}` here and more accurately specify the path to the go.mod file instead of the generic lockfile.

Reply:
Thoughts on this @arewm : Specify the package manager lock file relative to the package path. If the package manager lock file is located in the root of your repository, enter `gomod`. Otherwise, provide the path to the lock file in the JSON object format. For example, `{"type": "gomod", "path": "subpath/to/the/other/module"}`. Additionally, if you have multiple lock files, you can provide the path to the lock file in the JSON array format. For example, `[{"path": ".", "type": "pip"}, {"path": "subpath/to/other/module", "type": "pip"}]`. Here, `.` indicates that the package manager lock file is located in the root of your repository.

Reply:
Maybe others won't agree with me, but I think the "simplicity" of just specifying the package manager does not have enough benefit to complicate matters when describing both methods. Instead, we should just ignore the fact that the "default" case is there (users can read the docs and use it if they want) or they can specify the `type` and `path`.

If you want to include both, just introduce it "advanced" way as if it were the only way and include a `NOTE` for the special case of go.mod being in the root.

You can even add an additional note about multiple package managers/paths if you want.

Reply:
@chmeliik @brunoapimentel Please let us know your thoughts on @arewm suggestion. 
As for me, if what we have is accurate, has the correct flow, and conveys what we want to convey, I don't think we should change it. Having said that, I'm open to suggestions and updates.

Reply:
I tend to agree with @arewm here. While the `gomod` shorthand is nice when using `cachi2` as a CLI tool, it may only bring confusion for the RHTAP yaml.

It may be better to document only the JSON object and JSON array options